### ISSUE TITLE
Move K8sDatapathEgressGatewayTest tests to cilium-cli's connectivity tests running on ci-datapath

### ISSUE THREAD

**Issue by jibi** [cilium/cilium#24151]
This issue is about moving the egress gateway e2e tests from ginkgo to the new cilium-cli's datapath test suite.

The entire `K8sDatapathEgressGatewayTest` test suite lives in https://github.com/cilium/cilium/blob/master/test/k8s/egress.go.

This suite runs the same set of tests for each combination of the following configurations:
* tunneling: disabled/vxlan
* endpoint routes: disabled/enabled
* XDP: disabled/enabled

This might partially overlap with the different configurations that the `ci-datapath` workflow is running on: https://github.com/cilium/cilium/blob/f9d290c20ee9349da28bf8164ab8e7eb178079b0/.github/workflows/conformance-datapath.yaml#L121-L195 

Then, for each configuration, we run 3 group of tests:

* `no egress gw policy`: we run a generic connectivity test, both from the worker and the gateway node, with egress gateway enabled and without any policy in place
* `egress gw policy`: we run the same generic connectivity test together with a specific egress gateway test,  both from the worker and the gateway node, with egress gateway enabled and a few policies in place
* `egress gw policy with exclusion CIDRs`: similar to the previous one, but we specifically test the `excludedCIDRs` property

The generic connectivity test mentioned before: https://github.com/cilium/cilium/blob/666d6991d9d0a9759507737eaf0281e47671b802/test/k8s/egress.go#L177-L267

runs the following checks:
* TCP/UDP pod to pod connectivity works
* reply traffic for a connection initiated outside the cluster does not go through egress gateway (both nodeport services and pods)

The other egress gateway specific test mentioned before: https://github.com/cilium/cilium/blob/666d6991d9d0a9759507737eaf0281e47671b802/test/k8s/egress.go#L138-L175

tests that a connection from a pod matching an egress gateway policy is (or isn't if we are testing excluded CIDRs) SNATed through egress gateway

**Comment by brb** [comment]
> This might partially overlap with the different configurations that the ci-datapath workflow is running on

We can just enable the EGW in the existing configs.

**Comment by pchaigno** [comment]
Blocked by https://github.com/cilium/cilium/issues/23283 as the egress gateway requires BPF masquerading.

**Comment by lmb** [comment]
I think we'll need to add four more cases to the test matrix, since none of the existing cases have `loadBalancer.acceleration=testing-only`.

| workflow | tunnel   | endpoint-routes | loadBalancer.acceleration |
|----------|----------|-------------------|---------------------------|
| 7        | disabled | disabled          | disabled                  |
| 6        | vxlan    | disabled          | disabled                  |
| 5        | disabled | enabled           | disabled                  |
| 4        | vxlan    | enabled           | disabled                  |
|          | disabled | disabled          | testing-only              |
|          | enabled  | disabled          | testing-only              |
|          | disabled | enabled           | testing-only              |
|          | enabled  | enabled           | testing-only              |

Does that sound acceptable @brb?

**Comment by brb** [comment]
I'd suggest for now to ignore the XDP (`loadBalancer.acceleration=testing-only`), as it is currently blocked by https://github.com/cilium/cilium/issues/18846 (the first item).

**Comment by lmb** [comment]
```
unable to retrieve cilium status: container cilium-agent is in CrashLoopBackOff, exited with code 1: level=fatal msg="BPF masquerade is not supported for IPv6." subsys=daemon
```

Matrix 4-7 have IPv6 enabled, which conflicts with BPF masq, which is required by EGW.

**Comment by lmb** [comment]
Current status: I think that all pre-requisites are there to run egressgw tests on ci-datapath but I haven't done any work on porting tests to cilium-cli.

**Comment by jibi** [comment]
Quick update:
* basic infra and a first test case have been merged into the CLI: https://github.com/cilium/cilium-cli/pull/1555
* conformance-e2e test has been switched to v0.14.2 so now we are running this new test in `main` and `v1.13`: https://github.com/cilium/cilium/pull/25194

Next step will be porting the remaining test cases we have in the ginkgo tests
